cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -3896,6 +3896,41 @@ INSTANTIATE_TEST_SUITE_P(WriteDictionary, TestArrowWriteDictionary,
                          ::testing::Values(ParquetDataPageVersion::V1,
                                            ParquetDataPageVersion::V2));
 
+TEST(TestArrowWriteNestedDictionary, TestNullCount) {
+  std::shared_ptr<::arrow::Array> dictionary = ArrayFromJSON(::arrow::utf8(), R"(["z"])");
+  std::shared_ptr<::arrow::Array> indices =
+      ArrayFromJSON(::arrow::int32(), R"([0, null, 0])");
+  std::shared_ptr<::arrow::Array> offsets =
+      ArrayFromJSON(::arrow::int32(), R"([null, 0, null, null, null, 3])");
+  ASSERT_OK_AND_ASSIGN(std::shared_ptr<::arrow::Array> dict_encoded,
+                       ::arrow::DictionaryArray::FromArrays(indices, dictionary));
+  // values looks like [null, ["z", null, "z"], null, null, null]
+  ASSERT_OK_AND_ASSIGN(
+      std::shared_ptr<Array> values,
+      ::arrow::ListArray::FromArrays(*offsets, *dict_encoded, default_memory_pool()));
```

Comment:
You can directly create of list-of-dictionary array using `ArrayFromJSON`. For example:
```c++
auto type = list(dictionary(int32(), utf8()));
auto values = ArrayFromJSON(type, R"([null, ["z", null, "z"], null, null, null])");
```

Reply:
done.